### PR TITLE
DCOS-8052: Refactor FilterHeadline

### PR DESCRIPTION
**Pull request by daiyiwhale**
Bring `FilterHeadline` to es6 and add a new prop that forces showing the `clear` button for clearing the filters even if `n of n` objects are being displayed.

before 
doesn't show 'clear' menu when n of n objects are filtered

after
shows 'clear' menu when n of n objects are filtered

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @orlandohohmeier

File: src/js/components/FilterHeadline.js [hunk 1]
```diff
@@ -1,53 +1,46 @@
...
 
-var FilterHeadline = React.createClass({
+const METHODS_TO_BIND = [
+  'handleReset'
+];
 
-  displayName: 'FilterHeadline',
+class FilterHeadline extends React.Component {
+  constructor() {
+    super();
```
Comment (kennyt) [inline src/js/components/FilterHeadline.js]: let's do `...arguments`

File: src/js/components/FilterHeadline.js [hunk 2]
```diff
@@ -1,53 +1,46 @@
-var classNames = require('classnames');
-var React = require('react');
+import React, {PropTypes} from 'react';
+import classNames from 'classnames';
```
Comment (kennyt) [inline src/js/components/FilterHeadline.js]: :abc:

File: src/js/components/FilterHeadline.js [hunk 3]
```diff
@@ -1,53 +1,46 @@
...
-      'hidden': filteredLength !== totalLength
+      'hidden': (isFiltering == null && currentLength !== totalLength) || (isFiltering != null && isFiltering)
     });
 
-    var anchorClassSet = classNames({
+    let anchorClassSet = classNames({
       'h4 clickable': true,
       'inverse': inverseStyle,
-      'hidden': filteredLength === totalLength
+      'hidden': (isFiltering == null && currentLength === totalLength) || (isFiltering != null && !isFiltering)
```
Comment (kennyt) [inline src/js/components/FilterHeadline.js]: can we save this in a variable? we do the same exact thing on line 31

File: src/js/components/FilterHeadline.js [hunk 4]
```diff
@@ -23,24 +23,25 @@ class FilterHeadline extends React.Component {
 
   render() {
     let {currentLength, inverseStyle, isFiltering, name, totalLength} = this.props;
+    let hideFilteredClasses = (isFiltering == null && currentLength === totalLength) || (isFiltering != null && !isFiltering);
```
Comment (kennyt) [inline src/js/components/FilterHeadline.js]: this has to be longer than 80 :) sorry for the nits

Comment (daiyiwhale) [inline src/js/components/FilterHeadline.js]: haha no worries, I did the refactoring pretty hastily xD

File: src/js/pages/system/UnitsHealthDetail.js [hunk 5]
```diff
@@ -163,8 +163,9 @@ class UnitsHealthDetail extends mixin(StoreMixin) {
           title={unit.getTitle()} />
         <FilterHeadline
           currentLength={visibleData.length}
+          isFiltering={healthFilter !== 'all' || searchString !== ''}
```
Comment (kennyt) [inline src/js/pages/system/UnitsHealthDetail.js]: we use filterheadline in a lot more places than this right?

Comment (daiyiwhale) [inline src/js/pages/system/UnitsHealthDetail.js]: oops those changes were going to be a different PR xD

Comment (daiyiwhale) [inline src/js/pages/system/UnitsHealthDetail.js]: fixed

**Comment by orlandohohmeier** [comment]
LGTM! 🚢

**Comment by rcorral** [comment]
retest this please

File: src/js/components/FilterHeadline.js [hunk 6]
```diff
@@ -71,6 +67,21 @@ var FilterHeadline = React.createClass({
...
+
+FilterHeadline.defaultProps = {
+  inverseStyle: false
+};
+
+FilterHeadline.propTypes = {
+  currentLength: PropTypes.number.isRequired,
+  inverseStyle: PropTypes.bool,
+  // Optional prop used to force the "Clear" button to show even when n of n
+  // items are currenetly displayed.
```
Comment (kennyt) [inline src/js/components/FilterHeadline.js]: typo here

Comment (daiyiwhale) [inline src/js/components/FilterHeadline.js]: fixed!
